web/source/osk/visualKeyboard.ts

```diff
@@ -1371,9 +1404,20 @@ namespace com.keyman.osk {
         }
 
         keys[i]['sp'] = Processor.stateKeys[states[i]] ? Layouts.buttonClasses['SHIFT-ON'] : Layouts.buttonClasses['SHIFT'];
-        var btn = document.getElementById(layerId+'-'+states[i]);
+        let keyId = layerId+'-'+states[i]
+        var btn = document.getElementById(keyId);
 
-        this.setButtonClass(keys[i], btn, this.layout);
+        if(btn == null) {
+          //This can happen when using BuildDocumentationKeyboard, as the OSK isn't yet in the
+          //document hierarchy.  Sometimes.  (It's weird.)
+          btn = this.findKeyElement(layerId, states[i]);
+        }
+
+        if(btn != null) {
+          this.setButtonClass(keys[i], btn, this.layout);
+        } else {
+          console.warn("Could not find key to apply style: \"" + keyId + "\"");
```

Comment:
If `btn` is still null at this point, isn't the keyID `states[i]` (l.1413)?

Reply:
KMW key ids are of the form `<layer>-<key-name>`.  Note the definition of `keyId` earlier in the code, which uses `states[i]` in place of `<key-name>`.